mis_builder/models/aep.py

```diff
@@ -0,0 +1,291 @@
+
+        Domains and accounts are extracted and stored in the map
+        so when all expressions have been parsed, we know what to query.
+        """
+        for mo in self.ACC_RE.finditer(expr):
+            _, mode, account_codes, domain = self._parse_match_object(mo)
+            key = (domain, mode)
+            if account_codes:
+                self._map_account_ids[key].update(account_codes)
+            else:
+                self._set_all_accounts.add(key)
+
+    def done_parsing(self, account_domain):
+        # load account codes and replace account codes by account ids in _map
+        for key, account_codes in self._map_account_ids.items():
+            self._load_account_codes(account_codes, account_domain)
+            account_ids = set()
+            for account_code in account_codes:
+                account_ids.update(self._account_ids_by_code[account_code])
+            self._map_account_ids[key] = list(account_ids)
+
+    def get_aml_domain_for_expr(self, expr):
+        """ Get a domain on account.move.line for an expression.
+
+        Only accounting expression in mode 'p' and 'e' are considered.
+
+        Prerequisite: done_parsing() must have been invoked.
+
+        Returns a domain that can be used to search on account.move.line.
+        """
+        aml_domains = []
+        for mo in self.ACC_RE.finditer(expr):
+            field, mode, account_codes, domain = self._parse_match_object(mo)
+            if mode == MODE_INITIAL:
+                continue
+            aml_domain = list(domain)
+            if account_codes:
+                account_ids = set()
+                for account_code in account_codes:
+                    account_ids.update(self._account_ids_by_code[account_code])
+                aml_domain.append(('account_id', 'in', tuple(account_ids)))
+            if field == 'crd':
+                aml_domain.append(('credit', '>', 0))
+            elif field == 'deb':
+                aml_domain.append(('debit', '>', 0))
+            aml_domains.append(expression.normalize_domain(aml_domain))
+        return expression.OR(aml_domains)
+
+    def get_aml_domain_for_dates(self, date_start, date_end, mode):
+        if mode != MODE_VARIATION:
+            raise RuntimeError("")  # TODO
+        return [('date', '>=', date_start), ('date', '<=', date_end)]
+
+    def get_aml_domain_for_periods(self, period_start, period_end, mode):
+        period_obj = self.env['account.period']
+        move_obj = self.env['account.move']
+        domain_list = []
+        if mode == MODE_VARIATION:
+            compute_period_ids = period_obj.build_ctx_periods(
+                period_start.id,
+                period_end.id)
+            domain_list.extend([('period_id', 'in', compute_period_ids)])
+        else:
+            period_to = period_end
+            if mode == MODE_INITIAL:
+                # Processing to get the first period which isn't special
+                # before end period
```

Comment:
@adrienpeiffer  Ok, then is it not better to search on period directly, no need to search on account moves.

Also, what if reopening has been done and period_to is the first period of the fiscal year? Then the resulting domain must be the opening period only.